### PR TITLE
Lau/grid generator for heatmap

### PR DESCRIPTION
**Pull request by LauJosefsen**
Heatmap and improved ENC sidebar.

### CODE REVIEW

File: src/api/service/ais_data_service.py [hunk 1]
```diff
@@ -209,18 +208,60 @@ def __build_dict(cursor, row):
...
+        else:
+            search = f"%{search}%"
+
+        connection = psycopg2.connect(dsn=self.dsn)
+        cursor = connection.cursor()
+
+        query = """
+                SELECT * FROM (
+                    SELECT
+                    *,
```
Comment (Alexikik) [inline src/api/service/ais_data_service.py:226]: Hæsligt

File: src/frontend/src/App.tsx [hunk 2]
```diff
@@ -1,61 +1,53 @@
...
-                <>
-                  {settings.showEnc ? <MapEncCells limit={settings.encLimit} offset={settings.encOffset} search={settings.encSearch} /> : ''}
-                  {settings.showTrack ? <MapTrack limit={settings.trackLimit} offset={settings.trackOffset} search={settings.trackSearch} /> : ''}
-                </>
-              )}
+                        <SettingsContext.Consumer>
+                            {({ settings }) => (
+                                <>
+                                    {settings.showEnc ? <MapEncCells bounds={settings.encBounds} search={settings.encSearch} /> : ""}
+                                    {/* {settings.showTrack ? <MapTrack limit={settings.trackLimit} offset={settings.trackOffset} search={settings.trackSearch} /> : ''} */}
```
Comment (Alexikik) [inline src/frontend/src/App.tsx:38]: ?

Comment (Alexikik) [reply src/frontend/src/App.tsx:38]: why comment?

**Review by Alexikik — APPROVED** [review]
Mhh, mæh

**Review by Noobi99 — APPROVED** [review]

**Pull request merged**